include/seqan3/alignment/matrix/detail/alignment_score_matrix_proxy.hpp

```diff
@@ -0,0 +1,51 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::alignment_score_matrix_proxy.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/simd/concept.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief A proxy type for a unified access to the score matrix during alignment computation.
+ * \tparam score_type The type wrapped by this proxy.
+ * \ingroup alignment_matrix
+ *
+ * \details
+ *
+ * Provides named accessors to the respective values of the score matrix during the alignment computation.
+ * The `score_type` must either model the seqan3::Arithmetic or seqan3::detail::Simd vector concept.
+ * In some cases the last value to the left is stored in a different location then the next left value that
+ * will be used in the subsequent column (e.g. in the banded alignment matrix). Accordingly the left value is
+ * split in a reference for reading, referred to as `r_left` (for reading), and a reference to the next left value
+ * on the same row, referred to as `w_left` (for writing).
+ */
+template <typename score_type>
+struct alignment_score_matrix_proxy
```

Comment:
No this is not possible, since I have different requirements on the value type. alignment_trace requires the type to be a trace direction. At the end there would be 8 loc saved, 4 if I do inline documentations, which I will do 😄 . Not sure that this is worth it then.